sdk/bento/graph/ast.py

```diff
@@ -0,0 +1,82 @@
+#
+# Bentobox
+# SDK - Graph
+# AST Utils
+#
+
+
+import gast
+
+from tempfile import NamedTemporaryFile
+from astunparse import unparse
+from importlib.util import spec_from_file_location, module_from_spec
+from gast import AST, Call, FunctionDef, keyword, Name, Load, Attribute
+from inspect import getsource, cleandoc
+from textwrap import dedent
+from typing import Any, Dict, Optional
+
+
+def parse_ast(obj: Any) -> AST:
+    """Parse the AST of the given `obj`.
+
+    Args:
+        obj: Object to obtain AST for. Can be any object supported by `inspect.getsource()`.
+
+    Returns:
+        The AST parsed from the the given `obj`
+    """
+    src = getsource(obj)
+    clean_src = dedent((src))
+    return gast.parse(clean_src)
+
+
+def call_func_ast(
+    fn: FunctionDef, args: Dict[str, Any], attr_parent: Optional[str] = None
+) -> Call:
+    """Call the Function with the given FunctionDef AST  with the given args.
+
+    Applies the arguments in `args` by argument name in to `fn` and calls the
+    function by creating a Call AST node. Any extra arguments in `args` not used
+    to apply arguments in `fn` is ignored.
+
+    Args:
+        fn: FunctionDef AST of the function to call.
+        args: Mapping of argument name to argument value to apply function call.
+        attr_parent: Optionally specify name of the parent attribute required
+            to reference the given `fn`. Genrates a call with `attr_parent.fn(...)`.
+    Returns:
+        Call AST with the given args applied that represents the function call.
+    """
+    # collect names of parameters from fn_call
+    param_names = set([param.id for param in fn.args.args])
+    # extract arguments and apply to params
+    apply_arg = {name: value for name, value in args.items() if name in param_names}
```

Comment:
> I think you can reverse the way you create apply_args here. Instead of checking if args is in fn.args.args, check if fn.args.args is in args:

 Agreed. This function is simplified in the next PR.

Reply:
Ok, I'll look at the next PR and resolve this for now